contracts/InterchainTokenService.sol

```diff
@@ -400,6 +433,9 @@ contract InterchainTokenService is
         } else {
             if (chainNameHash == keccak256(bytes(destinationChain))) revert CannotDeployRemotelyToSelf();
 
+            // Make sure that the token manager already exists on this chain.
+            deployedTokenManager(tokenId);
```

Comment:
this is already checked in the below method

Reply:
nice catch